### PR TITLE
Slime renaming potion

### PR DESCRIPTION
**Pull request by coiax**
🆑 coiax
add: By injecting a light pink slime extract with water, you can create
a renaming potion, which allows a sentient being to change the name it
presents to the world.
add: The slime person minor reaction for light pink slime extract now produces a renaming potion, rather than a docility potion.
/🆑

- This does not change your ID, or your datacore records, or anyone's traitor objectives. See the implementation of `fully_replace_character_name()`.
- Renaming potions have the renaming option presented to the **target**, so they are in control of their name. You cannot rename the captain to Captain Smells-like-butt in a drive by attack.
- For reference, mulligan toxin is created with slime toxin + mutagen.
- Renaming potions will be available at the same time that sentience potions are available.

@XDTM

### CODE REVIEW

**Comment by fludd12** [comment]
This has the added benefit of the pink slime line being able to change gender and name, letting players have neat, Ranma style alter-egos.

**Comment by PKPenguin321** [comment]
I miss hand labelers working on mobs

**Comment by XDTM** [comment]
As i mentioned in irc, i believe the potion should only rename animals as humans and golems can either use mulligan, go to genetics, or have a plastic surgery

**Comment by fludd12** [comment]
I mean, you could do that, but then you'd get one of the... _interesting_ pre-genned names.

**Comment by chesse20** [comment]
i l;ove this so much i use it and then rename myself DanTDM LOL!!

File: code/modules/reagents/chemistry/recipes/slime_extracts.dm [hunk 1]
```diff
@@ -468,6 +468,18 @@
 	new /obj/item/slimepotion/slime/sentience(get_turf(holder.my_atom))
 	..()
 
+/datum/chemical_reaction/slime/renaming
+	name = "Renaming Potion"
+	id = "m_renaming_potion"
+	required_container = /obj/item/slime_extract/lightpink
+	required_reagents = list("water" = 1)
+	required_other = 1
```
Comment (Cyberboss) [inline code/modules/reagents/chemistry/recipes/slime_extracts.dm]: TRUE

File: code/modules/research/xenobiology/xenobiology.dm [hunk 2]
```diff
@@ -825,6 +825,39 @@
...
+		return
+	if(!M.ckey) //only works on animals that aren't player controlled
+		to_chat(user, "<span class='warning'>[M] is not self aware, and cannot pick its own name.</span>")
+		return
+
+	being_used = TRUE
+
+	var/new_name = stripped_input(M, "What would you like your name to be?", "Input a name", M.real_name, MAX_NAME_LEN)
+
+	if(!new_name || QDELETED(src) || QDELETED(M) || new_name == M.real_name)
```
Comment (Cyberboss) [inline code/modules/research/xenobiology/xenobiology.dm]: check adjacency

File: code/modules/research/xenobiology/xenobiology.dm [hunk 3]
```diff
@@ -825,6 +825,39 @@
...
+		being_used = FALSE
+		return
+
+	M.visible_message("<span class='notice'><span class='name'>[M]</span> has a new name, <span class='name'>[new_name]</span>.</span>", "<span class='notice'>Your old name of <span class='name'>[M.real_name]</span> fades away, and your new name <span class='name'>[new_name]</span> anchors itself in your mind.</span>")
+	message_admins("[ADMIN_LOOKUPFLW(M)] used [src] to rename themselves from <span class='name'>[M.real_name]</span> to <span class='name'>[new_name]</span>.")
```
Comment (Cyberboss) [inline code/modules/research/xenobiology/xenobiology.dm]: themselves?

File: code/modules/research/xenobiology/xenobiology.dm [hunk 4]
```diff
@@ -825,6 +825,39 @@
 	L.regenerate_icons()
 	qdel(src)
 
+/obj/item/slimepotion/slime/renaming
+	name = "renaming potion"
+	desc = "A potion that allows a self-aware being to change what name it subconciously presents to the world."
```
Comment (Cyberboss) [inline code/modules/research/xenobiology/xenobiology.dm]: But you can use it on other mobs?

File: code/modules/reagents/chemistry/recipes/slime_extracts.dm [hunk 5]
```diff
@@ -468,6 +468,18 @@
...
+
+/datum/chemical_reaction/slime/renaming/on_reaction(datum/reagents/holder)
+	new /obj/item/slimepotion/slime/renaming(get_turf(holder.my_atom))
```
Comment (Cyberboss) [inline code/modules/reagents/chemistry/recipes/slime_extracts.dm]: holder.my_atom.drop_location()

**Review by Cyberboss — DISMISSED** [review]

File: code/modules/research/xenobiology/xenobiology.dm (hunk 3 above)
Comment (coiax) [reply code/modules/research/xenobiology/xenobiology.dm]: Well, the name is chosen by the person that the potion is affecting, not the person who used the potion.

File: code/modules/research/xenobiology/xenobiology.dm (hunk 4 above)
Comment (coiax) [reply code/modules/research/xenobiology/xenobiology.dm]: Yes?

File: code/modules/research/xenobiology/xenobiology.dm (hunk 3 above)
Comment (XDTM) [reply code/modules/research/xenobiology/xenobiology.dm]: A clearer wording might be "[user] used [src] on [target], letting them rename themselves into [newname]."

File: code/modules/research/xenobiology/xenobiology.dm (hunk 2 above)
Comment (coiax) [reply code/modules/research/xenobiology/xenobiology.dm]: but TK gave me nightmares

**Comment by ShizCalev** [comment]
doesn't compile.

**Comment by coiax** [comment]
Having a bad week, I'll reopen when I'm ready to finish this.

File: code/modules/research/xenobiology/xenobiology.dm [hunk 6]
```diff
@@ -912,6 +912,38 @@
 	L.regenerate_icons()
 	qdel(src)
 
+<<<<<<< HEAD
```
Comment (Cyberboss) [inline code/modules/research/xenobiology/xenobiology.dm]: .

**Comment by coiax** [comment]
I don't understand why it's not compiling...

File: code/modules/research/xenobiology/xenobiology.dm [hunk 7]
```diff
@@ -912,6 +912,37 @@
...
+	being_used = TRUE
+
+	var/new_name = stripped_input(M, "What would you like your name to be?", "Input a name", M.real_name, MAX_NAME_LEN)
+
+	if(!new_name || QDELETED(src) || QDELETED(M) || new_name == M.real_name || !M.Adjacent(user))
+		being_used = FALSE
+		return
+
+	M.visible_message("<span class='notice'><span class='name'>[M]</span> has a new name, <span class='name'>[new_name]</span>.</span>", "<span class='notice'>Your old name of <span class='name'>[M.real_name]</span> fades away, and your new name <span class='name'>[new_name]</span> anchors itself in your mind.</span>")
+	message_admins("[ADMIN_LOOKUPFLW(M)] used [src] on [ADMIN_LOOKUPFLW(target)], letting them rename themselves into [newname].")
```
Comment (XDTM) [inline code/modules/research/xenobiology/xenobiology.dm]: replace target and newname with M and new_name

**Comment by ExcessiveUseOfCobblestone** [comment]
@PKPenguin321 they still should, do they not?

**Comment by ExcessiveUseOfCobblestone** [comment]
@fludd12 yes, that is purposeful to avoid legal-changeling which you should do at genetics or whathaveyou

**Comment by coiax** [comment]
Yeah, I tested it, it works.

File: code/modules/research/xenobiology/xenobiology.dm (hunk 4 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/research/xenobiology/xenobiology.dm]: Self-Aware is confusing considering L926 disallows sentienced name-changing

File: code/modules/research/xenobiology/xenobiology.dm [hunk 8]
```diff
@@ -912,6 +912,39 @@
...
+	desc = "A potion that allows a self-aware being to change what name it subconciously presents to the world."
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "potgreen"
+
+	var/being_used = FALSE
+
+/obj/item/slimepotion/slime/renaming/attack(mob/living/M, mob/user)
+	if(being_used || !ismob(M))
+		return
+	if(!M.ckey) //only works on animals that aren't player controlled
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/research/xenobiology/xenobiology.dm]: `//only works on animals that ARE player controlled`

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]

**Comment by ExcessiveUseOfCobblestone** [comment]
I'm not a fan of carbons getting this, I would prefer a special form of plastic surgery (techwebs) that allows the user to pick any name as opposed to the preset ones... or just use genetics you lazy bums.

**Comment by optimumtact** [comment]
I don't like this

**Comment by coiax** [comment]
Why?

**Comment by optimumtact** [comment]
I dunno I just do

**Pull request merged**